### DISCUSSION TITLE
What are the networking features?

### DISCUSSION THREAD

**Discussion by YoukaiTheGit** [stepmania/stepmania#2180] [Q&A]
In the game, there's an option to connect to a server.  What is this for?  Online multiplayer, community scoreboard, remote input devices... ?  Thanks.

**Comment by namgo** [comment]
Hi! I dug into this briefly and as this comment has gone without a response for quite some time I figured I'd give a (very non-definite response - I'm not a SM developer).

Stepmania contains code to connect to a server but the base repository doesn't seem to have a working implementation of a server upon which to play. Stepmania online appears to have (or had?) an implementation where you can play if you log onto their website ^1

There is a python implementation and someone working on a rust one, that I have not personally tested or used ^2 and ^3

I'm not sure if this is what you're looking for, and frankly I'm not sure if I should even be writing a response as my own response is a non-answer; stepmania has the ability to do online multiplayer but not a lot of people seem to host servers - it would seem.

[1] https://www.reddit.com/r/Stepmania/comments/138ofc/what_server_to_play_on/
[2] https://github.com/Moutix/stepmania-server
[3] https://www.youtube.com/watch?v=AASvq0wpSSQ
